packages/perseus-core/src/parse-perseus-json/perseus-parsers/radio-widget.ts

```diff
@@ -47,3 +53,107 @@ export const parseRadioWidget: Parser<RadioWidget> = parseWidget(
+                clue: optional(string),
+                correct: optional(boolean),
+                isNoneOfTheAbove: optional(boolean),
+                // deprecated
+                // There is an import cycle between radio-widget.ts and
+                // widgets-map.ts. The anonymous function below ensures that we
+                // don't refer to parseWidgetsMap before it's defined.
+                widgets: optional((rawVal, ctx) =>
+                    parseWidgetsMap(rawVal, ctx),
+                ),
+            }),
+        ),
+        hasNoneOfTheAbove: optional(boolean),
+        countChoices: optional(boolean),
+        randomize: optional(boolean),
+        multipleSelect: optional(boolean),
+        deselectEnabled: optional(boolean),
+        // deprecated
+        onePerLine: optional(boolean),
+        // deprecated
+        displayCount: optional(any),
+        // v0 props
+        // `noneOfTheAbove` is still in use (but only set to `false`).
+        noneOfTheAbove: optional(constant(false)),
+    }),
+);
+
+function migrateV1ToV2(
+    widget: ParsedValue<typeof parseRadioWidgetV1>,
+): RadioWidget {
+    const {options} = widget;
+    return {
+        ...widget,
+        version: {major: 2, minor: 0},
+        options: {
+            ...options,
+            numCorrect: deriveNumCorrect(options),
+        },
+    };
+}
+
+const version0 = optional(object({major: constant(0), minor: number}));
+const parseRadioWidgetV0: Parser<RadioWidget> = parseWidgetWithVersion(
+    version0,
+    constant("radio"),
+    object({
+        choices: array(
+            object({
+                content: defaulted(string, () => ""),
+                clue: optional(string),
+                correct: optional(boolean),
+                isNoneOfTheAbove: optional(boolean),
+                // deprecated
+                // There is an import cycle between radio-widget.ts and
+                // widgets-map.ts. The anonymous function below ensures that we
+                // don't refer to parseWidgetsMap before it's defined.
+                widgets: optional((rawVal, ctx) =>
+                    parseWidgetsMap(rawVal, ctx),
+                ),
+            }),
+        ),
+        hasNoneOfTheAbove: optional(boolean),
+        countChoices: optional(boolean),
+        randomize: optional(boolean),
+        multipleSelect: optional(boolean),
+        deselectEnabled: optional(boolean),
+        // deprecated
+        onePerLine: optional(boolean),
+        // deprecated
+        displayCount: optional(any),
+        // v0 props
+        // `noneOfTheAbove` is still in use (but only set to `false`).
+        noneOfTheAbove: optional(constant(false)),
+    }),
+);
+
+function migrateV0ToV1(
+    widget: ParsedValue<typeof parseRadioWidgetV1>,
+): RadioWidget {
+    const {options} = widget;
+    return {
+        ...widget,
+        version: {major: 1, minor: 0},
+        options: {
+            ...options,
+        },
+    };
```

Comment:
Do we even need to copy the options here?

```suggestion
    return {
        ...widget,
        version: {major: 1, minor: 0},
    };
```

Alternatively, maybe we should remove `noneOfTheAbove` and set `hasNoneOfTheAbove` to `false`, like in the original upgrade function?